File: src/main/dotnet/app/domain/feature/handlers/AddMessageHandler.cs
using System;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using MediatR;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;
using RabbitMQ.Client;
using se.clouds.javanet.app.domain.feature.commands;

public class AddMessageHandler : IRequestHandler<AddStartupMessage, Unit>
{
//connection is thread safe, use one channel per thread
    private IConfiguration _conf {get; set;}
    private readonly ILogger<AddMessageHandler> _logger;
    private readonly IMediator _mediator;
    public AddMessageHandler(ILogger<AddMessageHandler> logger, IMediator mediator, IConfiguration conf)
    {
        _conf=conf;
        _logger = logger;
        _mediator = mediator;
    }
        //public TimeSpan CachedTime { get; set; } = TimeSpan.FromHours(12);
    public Task<Unit> Handle(AddStartupMessage request, CancellationToken cancellationToken)
    {
        var factory = new ConnectionFactory() { HostName = _conf["Rabbit:Host"] };
        using(var connection = factory.CreateConnection())
        using(var channel = connection.CreateModel())
        {
            channel.QueueDeclare(queue: "hello",
                                 durable: false,
                                 exclusive: false,
                                 autoDelete: false,
                                 arguments: null);

            string message = "Hello World!";
            var body = Encoding.UTF8.GetBytes(message);

            channel.BasicPublish(exchange: "",
                                 routingKey: "hello",
                                 basicProperties: null,
                                 body: body);
            Console.WriteLine(" [x] Sent {0}", message);
        }
        return Unit.Task;
        //Console.WriteLine(" Press [enter] to exit.");
        //Console.ReadLine();
    }
}

File: src/main/java/se/clouds/javanet/app/domain/feature/handler/AddMessageHandler.java
package se.clouds.javanet.app.domain.feature.handler;

import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.ConnectionFactory;

import se.clouds.javanet.app.domain.feature.command.AddStartupMessage;
import se.clouds.javanet.core.configuration.Configuration;
import se.clouds.javanet.core.controller.ActionResult;
import se.clouds.javanet.core.controller.IActionResult;
import se.clouds.javanet.core.controller.ResultStatus;
import se.clouds.javanet.core.di.Di;
import se.clouds.javanet.core.mediator.IRequestHandler;
import se.clouds.javanet.core.mediator.MediatR;
import se.clouds.javanet.core.mediator.Task;

public class AddMessageHandler implements IRequestHandler<AddStartupMessage, IActionResult>
{
    private MediatR<IActionResult> mediatr = (MediatR)Di.GetMediator();

    public AddMessageHandler()
    {
        mediatr.addRequestHandler(new AddStartupMessage(), new HandlerTask());
    }

    public class HandlerTask implements Task<IActionResult>
    {
        private final static String QUEUE_NAME = "hello";
        @Override
        public IActionResult call() throws Exception {
            var result = new ActionResult();
            var host = Configuration.Get("Rabbit", "Host");

            ConnectionFactory factory = new ConnectionFactory();
            factory.setHost(host);
            try (Connection connection = factory.newConnection(); Channel channel = connection.createChannel()) {
                channel.queueDeclare(QUEUE_NAME, false, false, false, null);
                String message = "Hello World!";
                channel.basicPublish("", QUEUE_NAME, null, message.getBytes("UTF-8"));
                //System.out.println(" [x] Sent '" + message + "'");
            }
            result.SetStatus(ResultStatus.Status.OK);
            result.SetContent("");
            return result;
        }
    }
}
